core/src/data_sources/data_source.rs

```diff
@@ -537,6 +531,10 @@ impl DataSource {
         &self.config
     }
 
+    pub fn default_embedder_config(&self) -> &EmbedderConfig {
```

Comment:
Can we drop the `default` ?

Reply:
yes

Reply:
Updated.

Reply:
and we'll have `shadow_write_embedder_config` in the future. It's just that `default` does not carry any real meaning here.

Reply:
Agree, it was mainly to avoid the confusion between the field name the function name.